### PR TITLE
Decouple batch size and number of negatives

### PR DESCRIPTION
**Pull request by stes**
This PR adds a new argument, `num_negatives`, to both the `Loader` and `CEBRA` classes (torch and sklearn API). This allows to stabilize training behavior by providing additional negative examples to the InfoNCE loss independent of the batch size. We leveraged this logic for the models trained in [DCL](https://openreview.net/forum?id=ONfWFluZBI).

### Behavior

If the `num_negatives = None` (the default), the previous behavior is obtained for backwards compatibility and `Loader.batch_size` negative examples are drawn. If a different value is set, then the number of negative examples in all loaders will be set to `num_negatives` instead of `Loader.batch_size`.

The goodness of fit computation was also adapted to use the `num_negatives`.

### API modification

While implementing this functionality, I noticed an inconsistency between `single_session` and `multi_session` solvers. In the single session, we passed `self.batch_size` through the `get_indices` function, while in `multi_session` we use `self.batch_size` directly. The 2nd behavior makes more sense in the context of the general class design. I deprecated passing the `num_samples` parameter and adapted the samplers accordingly.

### CODE REVIEW

**Comment by stes** [comment]
Sorry, just saw that this was not merged yet. Picking this up again now.
